pkg/osquery/tables/wlan/psrunner.go

```diff
@@ -0,0 +1,104 @@
+package wlan
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+type PowerShell struct {
+	powerShell string
+}
+
+// New create new session
+func New() *PowerShell {
+	ps, _ := exec.LookPath("powershell.exe")
+	return &PowerShell{
+		powerShell: ps,
+	}
+}
+
+func runPos(ctx context.Context, output *bytes.Buffer) error {
+	posh := New()
+	dir, err := ioutil.TempDir("", "nativewifi")
+	if err != nil {
+		return errors.Wrap(err, "creating nativewifi tmp dir")
+	}
+	defer os.RemoveAll(dir)
+
+	outputFile := filepath.Join(dir, "nativewificode.cs")
+	nativeCodeFile, err := os.Create(outputFile)
+	if err != nil {
+		return errors.Wrap(err, "creating file for native wifi code")
+	}
+	// probably not needed, line 34 covers this
+	defer os.Remove(nativeCodeFile.Name())
+
+	_, err = nativeCodeFile.WriteString(nativeWiFiCode)
+	if err != nil {
+		return errors.Wrap(err, "writing native code file")
+	}
+
+	tmpl, err := template.New("command").Parse(getBSSIDCommandTemplate)
+	if err != nil {
+		return errors.Wrap(err, "parsing template")
+	}
+	commandOpts := struct {
+		NativeCodePath string
+	}{NativeCodePath: nativeCodeFile.Name()}
+	var command bytes.Buffer
+	if err := tmpl.ExecuteTemplate(&command, "command", commandOpts); err != nil {
+		return errors.Wrap(err, "executing template")
+	}
+
+	err = posh.execute(output, command.String())
+	if err != nil {
+		// TODO: log, not printf
+		fmt.Printf("error: %s\n", err)
+	}
+
+	return err
+}
+
+func (p *PowerShell) execute(out *bytes.Buffer, args ...string) error {
+	args = append([]string{"-NoProfile", "-NonInteractive"}, args...)
+	cmd := exec.Command(p.powerShell, args...)
+
+	var stderr bytes.Buffer
+	cmd.Stdout = out
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		fmt.Printf("%s", stderr.String())
+	}
+	return err
+}
+
+const simplecommand = `
```

Comment:
This looks like it should be in a `_test.go` ?

Reply:
Oh not a bad idea. I would've just deleted it tbh, just hadn't cleaned this up yet.